### PR TITLE
CRUMBS

### PR DESCRIPTION
**Pull request by henrylay97**
Creates a standard record object SRCRUMBSResult to store the result of CRUMBS's BDT in the CAF files.

Links to main PR https://github.com/SBNSoftware/sbncode/pull/250

### CODE REVIEW

File: sbnanaobj/StandardRecord/SRCRUMBSResult.h [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+#include <vector>
+
+namespace caf {
+/// Representation of CRUMBS MVA Slice ID output
+class SRCRUMBSResult {
+  public:
+  SRCRUMBSResult();
+  ~SRCRUMBSResult() {}
+
+  float score;
```
Comment (etyley) [inline sbnanaobj/StandardRecord/SRCRUMBSResult.h:17]: Include units in comments (Ideally in doxygen comment strings with a brief description of each)

File: sbnanaobj/StandardRecord/classes_def.xml [hunk 2]
```diff
@@ -32,9 +32,8 @@
    <version ClassVersion="10" checksum="83320005"/>
   </class>
 
-  <class name="caf::SRSlice" ClassVersion="12">
-   <version ClassVersion="12" checksum="3783103871"/>
```
Comment (etyley) [inline sbnanaobj/StandardRecord/classes_def.xml:36]: I think the legacy versions are required for backwards compatibility?

**Review by etyley — COMMENTED** [review]
A few minor comments on documentation and backwards compatibility

File: sbnanaobj/StandardRecord/SRCRUMBSResult.cxx [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  , tpc_NuVertexY(-5.f)
+  , tpc_NuWeightedDirZ(-5.f)
+  , tpc_StoppingChi2CosmicRatio(-5.f)
+  , pds_FMTotalScore(-5.f)
+  , pds_FMPE(-5.f)
+  , pds_FMTime(-5.f)
+  , crt_TrackScore(-5.f)
+  , crt_HitScore(-5.f)
+  , crt_TrackTime(-5.f)
+  , crt_HitTime(-5.f)
```
Comment (cjbackhouse) [inline sbnanaobj/StandardRecord/SRCRUMBSResult.cxx:27]: These should all be NaNs and the like. The convention is NaNs where the value is literally uninitialized (indicating a bug in the CAFMaker code). -5 is for the case where we have purposely considered the value and there is nothing appropriate to put.

File: sbnanaobj/StandardRecord/SRCRUMBSResult.h [hunk 4]
```diff
@@ -0,0 +1,42 @@
+////////////////////////////////////////////////////////////////////////
+// \file    SRCRUMBSResult.h
+////////////////////////////////////////////////////////////////////////
+#ifndef SRCRUMBSRESULT_H
+#define SRCRUMBSRESULT_H
+
+
+#include <vector>
```
Comment (cjbackhouse) [inline sbnanaobj/StandardRecord/SRCRUMBSResult.h:8]: Unused?
```suggestion
```

File: sbnanaobj/StandardRecord/classes_def.xml (hunk 2 above)
Comment (cjbackhouse) [reply sbnanaobj/StandardRecord/classes_def.xml:36]: Agreed. Any version number that has been in `develop` should remain here.

File: sbnanaobj/StandardRecord/SRCRUMBSResult.h [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+  float tpc_NuVertexY;
+  float tpc_NuWeightedDirZ;
+  float tpc_StoppingChi2CosmicRatio;
+  float pds_FMTotalScore;
+  float pds_FMPE;
+  float pds_FMTime;
+  float crt_TrackScore;
+  float crt_HitScore;
+  float crt_TrackTime;
+  float crt_HitTime;
```
Comment (cjbackhouse) [inline sbnanaobj/StandardRecord/SRCRUMBSResult.h:35]: <broken record>

Can we make these names more CAF-y?

I would actually be a fan of making three smaller classes so these can be `tpc.`, `pds.`, `crt.`. Some of these would work out well in CAF style, e.g. `crt.tracktime`. Some of the longer ones might be a struggle.

File: sbnanaobj/StandardRecord/SRCRUMBSResult.cxx (hunk 3 above)
Comment (henrylay97) [reply sbnanaobj/StandardRecord/SRCRUMBSResult.cxx]: Okiedokie, done! I've used `std::numeric_limits<int>::max()` for the ints.

**Review by cjbackhouse — APPROVED** [review]
Thanks!

File: sbnanaobj/StandardRecord/SRCRUMBSResult.h (hunk 5 above)
Comment (henrylay97) [reply sbnanaobj/StandardRecord/SRCRUMBSResult.h]: Really liked the idea of separate objects, not so sure about all the lower case names for the tpc vars but have gone with it to match convention.

**Review by etyley — APPROVED** [review]

**Pull request merged**